## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 509411`
Commit: [`10ba57b3a28a24893fc488eabc2dd25d1b29b618`](https://github.com/NixOS/nixpkgs/commit/10ba57b3a28a24893fc488eabc2dd25d1b29b618) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/10ba57b3a28a24893fc488eabc2dd25d1b29b618..pull/509411/head))
Merge: [`29d2396ebfd063af1f6eede6ec3cc975bbba1549`](https://github.com/NixOS/nixpkgs/commit/29d2396ebfd063af1f6eede6ec3cc975bbba1549)

Logs: https://github.com/gonsolo/nixpkgs-review-gha/actions/runs/24318584546


---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>librelane</li>
    <li>librelane.dist</li>
    <li>openroad</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>openroad</summary>
<pre>module_dcalc                                   = 178.43 sec*proc (383 tests)
module_graph                                   =  23.41 sec*proc (110 tests)
module_liberty                                 = 236.27 sec*proc (1065 tests)
module_network                                 =  18.72 sec*proc (604 tests)
module_parasitics                              =  90.92 sec*proc (216 tests)
module_power                                   =  74.27 sec*proc (93 tests)
module_sdc                                     =  59.96 sec*proc (909 tests)
module_sdf                                     =  67.46 sec*proc (125 tests)
module_search                                  = 261.27 sec*proc (1881 tests)
module_spice                                   =  13.19 sec*proc (135 tests)
module_util                                    =   6.23 sec*proc (258 tests)
module_verilog                                 = 119.59 sec*proc (318 tests)
tcl                                            = 215.92 sec*proc (312 tests)

Total Test time (real) = 1320.97 sec

The following tests FAILED:
	3491 - StaDcalcTest.UnitDelayCalcGateDelayWithLoads (Subprocess aborted) cpp module_dcalc
Errors while running CTest
make: *** [Makefile:71: test] Error 8</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>librelane</li>
    <li>librelane.dist</li>
    <li>openroad</li>
  </ul>
</details>

---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>openroad</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>openroad</summary>
<pre>1 warning generated.
In file included from /nix/build/nix-14401-155629099/source/src/odb/src/db/dbScanListScanInstItr.cpp:9:
In file included from /nix/build/nix-14401-155629099/source/src/odb/src/db/dbScanInst.h:12:
In file included from /nix/build/nix-14401-155629099/source/src/odb/src/db/dbInst.h:12:
In file included from /nix/build/nix-14401-155629099/source/src/odb/include/odb/dbTypes.h:11:
In file included from /nix/build/nix-14401-155629099/source/src/odb/include/odb/geom.h:18:
In file included from /nix/build/nix-14401-155629099/source/src/odb/include/odb/isotropy.h:20:
/nix/build/nix-14401-155629099/source/src/utl/include/utl/Logger.h:173:5: warning: &#x27;sprintf&#x27; is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  173 |     sprintf(tool_id, &quot;%s-%04d&quot;, tool_names_[tool], id);
      |     ^
/nix/store/q2dccg26bm7bn6ia1q30qkl5jck7wwgb-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:180:1: note: &#x27;sprintf&#x27; has been explicitly marked deprecated here
  180 | __deprecated_msg(&quot;This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.&quot;)
      | ^
/nix/store/q2dccg26bm7bn6ia1q30qkl5jck7wwgb-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:218:48: note: expanded from macro &#x27;__deprecated_msg&#x27;
  218 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
      |                                                       ^
1 warning generated.
[ 13%] Built target db
[ 62%] Built target libabc
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/op/openroad/package.nix
+++ b/pkgs/by-name/op/openroad/package.nix
@@ -44,23 +44,15 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "openroad";
-  version = "26Q1";
+  version = "26Q2";
 
   src = fetchFromGitHub {
     owner = "The-OpenROAD-Project";
     repo = "OpenROAD";
     tag = finalAttrs.version;
     fetchSubmodules = true;
-    hash = "sha256-DMyoqDse9W6ahOajEINzFpgLsSKam/I1mQkRSSKepI8=";
+    hash = "sha256-dB9PfPlp6vZ9+Th8LJE65BW9YeuUL0G4JtjzQxg6UpQ=";
   };
-
-  patches = [
-    (fetchpatch {
-      name = "fix-openroad-commit-2a8b2c7.patch";
-      url = "https://github.com/The-OpenROAD-Project/OpenROAD/commit/2a8b2c7dcda87679a69df323b2ada5f3a21554ea.patch";
-      hash = "sha256-vgmVpr+vHbOd8UUUUyJ8sTKi0Y7CWYatF006WX4+zFI=";
-    })
-  ];
 
   nativeBuildInputs = [
     bison
@@ -103,6 +95,34 @@ stdenv.mkDerivation (finalAttrs: {
   ]
   ++ lib.optionals stdenv.hostPlatform.isLinux [ libx11 ]
   ++ lib.optionals stdenv.hostPlatform.isDarwin [ llvmPackages.openmp ];
+
+  patches = [
+    # Fix UB in OpenSTA dcalc tests: fake Pin* pointers dereference via
+    # PinIdLess comparator, crashing with GCC 15's hardened vector bounds check.
+    # https://github.com/The-OpenROAD-Project/OpenSTA/pull/346
+    (fetchpatch {
+      url = "https://github.com/gonsolo/OpenSTA/commit/0e40b4f8a1c4b6af7225da31cd88a1d29d8a04a2.patch";
+      hash = "sha256-W9USjqp/hL1s3w3nKVMo/a5aSkeQ4Lp7gqASbZSlo9Y=";
+      stripLen = 1;
+      extraPrefix = "src/sta/";
+    })
+    # Feature-test std::from_chars to fix aarch64-darwin build where
+    # libcxx marks from_chars unavailable (macOS 26.0).
+    # https://github.com/The-OpenROAD-Project/OpenSTA/commit/a5921d1ca
+    (fetchpatch {
+      url = "https://github.com/The-OpenROAD-Project/OpenSTA/commit/a5921d1ca964971ada83be2c7c65bb84504fe179.patch";
+      hash = "sha256-j9BneXSIya/euYiol16swmrFkXTDZNTQwq3tPFkCLH0=";
+      stripLen = 1;
+      extraPrefix = "src/sta/";
+    })
+    # Replace deprecated sprintf with snprintf in Logger::error.
+    # macOS Apple SDK 14.4+ marks sprintf as deprecated, breaking -Werror builds.
+    # https://github.com/The-OpenROAD-Project/OpenROAD/pull/10127
+    (fetchpatch {
+      url = "https://github.com/The-OpenROAD-Project/OpenROAD/commit/2a9191bc5b2841a0c357886a2a1bc3ac0fe5271a.patch";
+      hash = "sha256-lxFZvybfG0Qpg1TyKdfZhKLYI3DSCYDE54ta6EnDBDo=";
+    })
+  ];
 
   postPatch = ''
     patchShebangs etc/

